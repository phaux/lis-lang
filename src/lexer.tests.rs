use super::*;

#[test]
fn prog_simple() {
    let input = "let x = 5;";
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn assign_str_escaped() {
    let input = r#" s = "hello\nworld\twith\\quotes\"" "#;
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn assign_str_unclosed() {
    let input = r#" s = "hello "#;
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn assign_num() {
    let input = " x = 5.5 ";
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn assign_num_invalid() {
    let input = " x = 5.5.5 ";
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn comment_line() {
    let input = "let x = 1; // this is a comment\nprint x";
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn comment_block() {
    let input = "let x = /* block comment */ 1;";
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}
