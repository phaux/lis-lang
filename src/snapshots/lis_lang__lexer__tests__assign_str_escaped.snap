---
source: src/lexer.tests.rs
expression: tokens
---
[
    Token {
        sigil: Ident {
            name: "s",
        },
        offset: 1..2,
        pos: Pos {
            line: 0,
            col: 1,
        }..Pos {
            line: 0,
            col: 2,
        },
    },
    Token {
        sigil: Eq,
        offset: 3..4,
        pos: Pos {
            line: 0,
            col: 3,
        }..Pos {
            line: 0,
            col: 4,
        },
    },
    Token {
        sigil: Str {
            val: "hello\nworld\twith\\quotes\"",
        },
        offset: 5..35,
        pos: Pos {
            line: 0,
            col: 5,
        }..Pos {
            line: 0,
            col: 35,
        },
    },
]
